### DISCUSSION TITLE
Any way to get this plugin to replace netrw?

### DISCUSSION THREAD

**Discussion by fenneltheloon** [mikavilpas/yazi.nvim#129] [General]
I'd ideally like yazi in nvim to open when I run `nvim .` instead of netrw. Anyone have an idea as to how to do that?

**Comment by mikavilpas** [comment]
Yes! This can be done by setting `open_for_directories` 

https://github.com/mikavilpas/yazi.nvim#%EF%B8%8F%EF%B8%8F-advanced-configuration

I use it myself.

**Comment by fenneltheloon** [comment]
When setting this option on version 0.10.0 and launching using `nvim .` everything seems to break. The cursor appears to be looking at the default empty buffer underneath yazi, and I cannot get it to focus back into yazi. Launching yazi from a hotkey after neovim has started yields no issues. It may have something to do with 2cf6057, as when I launch yazi from an open file it opens in terminal mode instead of insert mode.

**Reply by mikavilpas** [reply]
Hmm this sounds like some other plugin or setting might be conflicting. Could you follow this guide to see if you can reproduce it? https://github.com/mikavilpas/yazi.nvim/blob/master/documentation/reproducing-issues.md

**Reply by fenneltheloon** [reply]
Sure can! The minimal config is also broken or I'm calling `setup()` incorrectly, `repro.lua` is attached.

```lua
-- You can use this file to reproduce an issue with your configuration.

---@module "yazi"
---@module "lazy"

-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify('./.repro', ':p')

-- set stdpaths to use .repro
for _, name in ipairs({ 'config', 'data', 'state', 'cache' }) do
	vim.env[('XDG_%s_HOME'):format(name:upper())] = root .. '/' .. name
end

-- bootstrap lazy
local lazypath = root .. '/plugins/lazy.nvim'
if not vim.loop.fs_stat(lazypath) then
	vim.fn.system({
		'git',
		'clone',
		'--filter=blob:none',
		'https://github.com/folke/lazy.nvim.git',
		lazypath,
	})
end
vim.opt.runtimepath:prepend(lazypath)
vim.g.mapleader = ' '

-- install the following plugins
---@type LazySpec
local plugins = {
	'folke/tokyonight.nvim',
	{
		'mikavilpas/yazi.nvim',
		dependencies = {
			'nvim-lua/plenary.nvim',
		},
		event = 'VeryLazy',
		keys = {
			{
				-- 👇 choose your own keymapping
				'<leader>fy',
				function()
					require('yazi').yazi()
				end,

				{ desc = 'Open the file manager' },
			},
		},
		---@type YaziConfig
		opts = {
			open_for_directories = true,
		},
	},
}
require('lazy').setup(plugins, {
	root = root .. '/plugins',
})

vim.cmd.colorscheme('tokyonight')
-- add anything else here
require("yazi").setup()
```

**Reply by mikavilpas** [reply]
Thanks for the effort! Usually it's very difficult to offer good help, but this makes it a lot easier 🙂 

If you remove the last line, does it work for you then?

```diff
diff --git a/repro.lua b/repro.lua
index 5315962..9d414a3 100644
--- a/repro.lua
+++ b/repro.lua
@@ -58,4 +58,3 @@ require('lazy').setup(plugins, {
 
 vim.cmd.colorscheme('tokyonight')
 -- add anything else here
-require('yazi').setup()
```


https://github.com/mikavilpas/yazi.nvim/assets/300791/beba9d64-63a2-4c84-b692-045077139548

**Reply by mikavilpas** [reply]
Note that it breaks for me too if I leave the final line in:


https://github.com/mikavilpas/yazi.nvim/assets/300791/64f96941-a53b-4b38-9c81-188238fdc52e

**Reply by fenneltheloon** [reply]
Removed final line and that fixed the `repro` config but not mine so I went hunting for something that I edited. The plugin must be left set to load on `event = VeryLazy` or bad things happen. Fixed!

**Reply by JumpyLionnn** [reply]
I have the same issue but the difference is that i am using packer and i need the setup function to load the config
is there another solution?

**Reply by mikavilpas** [reply]
I might have forgotten some of the details about this issue, but as long as you are able to run `require("yazi").setup()` before you enter a directory, it should work. Right now the `.setup()` has to be run to set up some autocmds that run when a directory is entered - I don't know how else to do it (yet).

**Reply by mikavilpas** [reply]
Can you add the setup call somewhere in your config? I don't really know anything about packer.nvim, but I might be able to help one small step at a time.

**Reply by JumpyLionnn** [reply]
this is my init.lua file
```lua
require("yazi").setup({
    open_for_directories = true,

    use_ya_for_events_reading = false,

    use_yazi_client_id_flag = false,

    highlight_groups = {
      -- NOTE: this only works if `use_ya_for_events_reading` is enabled, etc.
      hovered_buffer = nil,
    },

    floating_window_scaling_factor = 1,

    yazi_floating_window_winblend = 0,

    log_level = vim.log.levels.OFF,

    keymaps = {
      show_help = '<f1>',
      open_file_in_vertical_split = '<c-v>',
      open_file_in_horizontal_split = '<c-x>',
      open_file_in_tab = '<c-t>',
      grep_in_directory = '<c-s>',
      replace_in_directory = '<c-g>',
      cycle_open_buffers = '<tab>',
      copy_relative_path_to_selected_files = '<c-y>',
    },

    yazi_floating_window_border = 'rounded',

    clipboard_register = "*",

})

vim.keymap.set("n", "<leader>pv", require("yazi").yazi)

-- some other initialization
require("init")
```
even when its the first thing initialized in init.lua i have the same problem

**Reply by mikavilpas** [reply]
I get the same behaviour, so this is reproducible. Can you see if the following approach would work around the issue for you?

```lua
vim.api.nvim_create_autocmd('UiEnter', {
  callback = function()
    require('yazi').setup({
      open_for_directories = true,
    })
    vim.keymap.set('n', '<leader>pv', require('yazi').yazi)
  end,
})
```

My guess is that for the current yazi.nvim implementation to work, it needs to be loaded somehow "later" than other things might be happening. I believe this somehow fixes the loading order as this seems to resolve the issue for me when I tried it without lazy.nvim just now.

Ideally I would recommend switching over to lazy.nvim, but I do understand it might not be possible for everybody. I don't think the loading order is going to change much in the future though, but I might add or change the dependencies. If you get any issues, feel free to send me a message.

**Reply by JumpyLionnn** [reply]
this way it works, thank you!

**Reply by sunbearc22** [reply]
@mikavilpas  Appending `vim.g.loaded_netrwPlugin = 1` to the above suppresses netrw too.  I like to suggest  that the [Without a package manager](https://github.com/mikavilpas/yazi.nvim#without-a-package-manager) section in your README.md be updated with:

```
vim.g.loaded_netrwPlugin = 1
vim.api.nvim_create_autocmd('UiEnter', {
  callback = function()
    require('yazi').setup({
      open_for_directories = true,
    })
    vim.keymap.set('n', '<leader>pv', require('yazi').yazi)
  end,
})
```

**Reply by mikavilpas** [reply]
Good idea, opened this to document it https://github.com/mikavilpas/yazi.nvim/pull/1289

But I'm getting rate limited right now, not sure why. Will retry later.
